js/client/modules/@arangodb/arango-collection.js

```diff
@@ -145,13 +93,26 @@ let appendOverwriteModeParameter = function (url, mode) {
 // / @brief append some boolean parameter to a URL
 // //////////////////////////////////////////////////////////////////////////////
 
-let appendBoolParameter = function (url, name, val) {
-  if (url.indexOf('?') === -1) {
-    url += '?';
-  }else {
-    url += '&';
+let appendBoolParameter = function (url, name, val, onlyIfSet = false) {
+  if (!onlyIfSet || (val !== undefined && val !== null)) {
+    if (url.includes('?')) {
+      url += '&';
+    } else {
+      url += '?';
+    }
+    url += name + (val ? '=true' : '=false');
+  }
+  return url;
+};
+
+// //////////////////////////////////////////////////////////////////////////////
+// / @brief append the waitForSync parameter to a URL
+// //////////////////////////////////////////////////////////////////////////////
+
+let appendSyncParameter = function (url, waitForSync) {
+  if (waitForSync) {
+    url = appendBoolParameter(url, 'waitForSync', waitForSync);
   }
-  url += name + (val ? '=true' : '=false');
   return url;
 };
```

Comment:
We have the exact same function in arango-database.js. Can we move this to some helpers file?

Reply:
in general, yes. But I don't want to make this PR even larger as it is already. I am happy to do a post-PR cleanup of some of the stuff you mentioned in the review comments, but I would like to decouple it from this PR.